src/server/internalServer.cpp

```diff
@@ -401,36 +400,25 @@ std::unique_ptr<Response> InternalServer::handle_meta(const RequestContext& requ
     return Response::build_404(*this, request, bookName, "");
   }
 
-  std::string content;
-  std::string mimeType = "text";
-
-  if (meta_name == "title") {
-    content = getArchiveTitle(*archive);
-  } else if (meta_name == "description") {
-    content = getMetaDescription(*archive);
-  } else if (meta_name == "language") {
-    content = getMetaLanguage(*archive);
-  } else if (meta_name == "name") {
-    content = getMetaName(*archive);
-  } else if (meta_name == "tags") {
-    content = getMetaTags(*archive);
-  } else if (meta_name == "date") {
-    content = getMetaDate(*archive);
-  } else if (meta_name == "creator") {
-    content = getMetaCreator(*archive);
-  } else if (meta_name == "publisher") {
-    content = getMetaPublisher(*archive);
-  } else if (meta_name == "favicon") {
-    getArchiveFavicon(*archive, 48, content, mimeType);
-  } else if (const unsigned illustrationSize = parseIllustration(meta_name)) {
-    getArchiveFavicon(*archive, illustrationSize, content, mimeType);
-  } else {
-    return Response::build_404(*this, request, bookName, "");
+  try {
+    auto meta_item = archive->getMetadataItem(meta_name);
+    auto response = ContentResponse::build(*this, meta_item.getData(), meta_item.getMimetype());
+    response->set_cacheable();
+    return std::move(response);
+  } catch (const zim::EntryNotFound& e) {
+    // Some (old) zim have no Illustration metadata but has a favicon in the content.
+    // But the api will return the favicon as a 48px Illustration. And so kiwix-serve
+    // will try to access 48px illustration even if we actually have a favicon.
+    if (const unsigned illustrationSize = parseIllustration(meta_name)) {
+      std::string content, mimetype;
+      getArchiveFavicon(*archive, illustrationSize, content, mimetype);
+      auto response = ContentResponse::build(*this, content, mimetype);
+      response ->set_cacheable();
+      return std::move(response);
+    } else {
+      return Response::build_404(*this, request, bookName, "");
+    }
```

Comment:
I think that this piece of code can be rewritten, by expanding `getArchiveFavicon()`, as follows:

```c++
  zim::Item meta_item;
  try {
    if (const unsigned illustrationSize = parseIllustration(meta_name)) {
      // Some (old) zim have no Illustration metadata but has a favicon in the content.
      // But the api will return the favicon as a 48px Illustration. And so kiwix-serve
      // will try to access 48px illustration even if we actually have a favicon.
      meta_item = archive.getIllustrationItem(size);
    } else {
      meta_item = archive->getMetadataItem(meta_name);
    }
  } catch (const zim::EntryNotFound& e) {
    return Response::build_404(*this, request, bookName, "");
  }
  auto response = ContentResponse::build(*this, meta_item.getData(), meta_item.getMimetype());
  response->set_cacheable();
  return std::move(response);
```

It may also make sense to extract the body of the `try` statement into a free helper function `zim::Item getIllustrationOrMetadataItem(const std::string& meta_name)`. Then you can restore the backward compatibility of the API (handling of the previously lower-case names of metadata entries as well as the now dropped `favicon` metadata without introducing any clutter to `InternalServer::handle_meta()`.

Reply:
BTW, the code in my previous comment should also address @rgaudin's [concern](https://github.com/kiwix/libkiwix/pull/626#pullrequestreview-791864405)